client/state/simple-payments/product-list/actions.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Internal dependencies
+ */
+import {
+	SIMPLE_PAYMENTS_PRODUCTS_LIST_RECEIVE,
+	SIMPLE_PAYMENTS_PRODUCTS_LIST_RECEIVE_UPDATE,
+	SIMPLE_PAYMENTS_PRODUCTS_LIST_RECEIVE_DELETE,
+} from 'state/action-types';
+
+export function receiveProductsList( siteId, numOfProducts, posts ) {
```

Comment:
Ah ic, so instead of dispatching `SIMPLE_PAYMENTS_PRODUCTS_LIST_RECEIVE_UPDATE` we can let the reducer directly listen to `SIMPLE_PAYMENTS_PRODUCTS_LIST_ADD`. With the data-layer, http actions are dispatched as side-effects. We're really interested in the add action, and we can always correct ourselves on failure afterwards if the request is not correct.